universe/supplycommit/supply_commit_states.go

```diff
@@ -0,0 +1,553 @@
+package supplycommit
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/mssmt"
+	"github.com/lightninglabs/taproot-assets/proof"
+	"github.com/lightninglabs/taproot-assets/tapsend"
+	"github.com/lightninglabs/taproot-assets/universe"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/protofsm"
+)
+
+var (
+	// ErrInvalidStateTransition is returned when we receive an unexpected
+	// event for a given state.
+	ErrInvalidStateTransition = fmt.Errorf("invalid state transition")
+
+	ErrInvalidState = fmt.Errorf("invalid state")
+)
+
+// Event is a special interface used to create the equivalent of a sum-type, but
+// using a "sealed" interface. Supply commit events can be used as input to
+// trigger a state transition, and also as output to trigger a new set of events
+// into the very same state machine.
+type Event interface {
+	eventSealed()
+}
+
+// Events is a special type constraint that enumerates all the possible protocol
+// events. This is used mainly as type-level documentation, and may also be
+// useful to constraint certain state transition functions.
+type Events interface {
+}
+
+// StateTransition is the StateTransition type specific to the supply commit
+// state machine.
+type StateTransition = protofsm.StateTransition[Event, *Environment]
+
+// State is our sum-type ish interface that represents the current universe
+// commitment state.
+type State interface {
+	// stateSealed is a special method that is used to seal the interface
+	// (only types in this package can implement it).
+	stateSealed()
+
+	// IsTerminal returns true if the target state is a terminal state.
+	IsTerminal() bool
+
+	// ProcessEvent takes a supply event event, and implements a state
+	// transition for the state.
+	//
+	// nolint:lll
+	ProcessEvent(Event, *Environment) (*StateTransition, error)
+
+	// String returns the name of the state.
+	String() string
+}
+
+// StateMachine is a state machine that handles creating and updating the
+// on-chain universe supply commitment for a given asset.
+//
+// nolint:lll
+type StateMachine = protofsm.StateMachine[Event, *Environment]
+
+// Config is a configuration struct that is used to initialize a new supply
+// commit state machine.
+type Config = protofsm.StateMachineCfg[Event, *Environment]
```

Comment:
I don't think these are used and they might be the only reason for the `lnd` replace in `go.mod`. I'm not sure.

Reply:
So the state machine isn't instantiated externally yet, once it is these will  be used. 

The lnd fork dep is needed to we can access the block block after confirmation, and also pass that into the state machine.